src/components/remote_control/test/commands/set_interior_vehicle_data_request_test.cc

```diff
@@ -0,0 +1,411 @@
+      , mock_app_(utils::MakeShared<NiceMock<MockApplication> >())
+      , rc_app_extention_(
+            utils::MakeShared<remote_control::RCAppExtension>(kModuleId)) {
+    ON_CALL(*mock_app_, app_id()).WillByDefault(Return(app_id_));
+    ON_CALL(mock_module_, resource_allocation_manager())
+        .WillByDefault(ReturnRef(mock_allocation_manager_));
+    ON_CALL(mock_module_, service()).WillByDefault(Return(mock_service_));
+    ON_CALL(*mock_service_, GetApplication(app_id_))
+        .WillByDefault(Return(mock_app_));
+    EXPECT_CALL(mock_module_, event_dispatcher())
+        .WillRepeatedly(ReturnRef(event_dispatcher_));
+    ServicePtr exp_service(mock_service_);
+    mock_module_.set_service(exp_service);
+  }
+
+  remote_control::request_controller::MobileRequestPtr CreateCommand(
+      application_manager::MessagePtr msg) {
+    return remote_control::RCCommandFactory::CreateCommand(msg, mock_module_);
+  }
+
+  application_manager::MessagePtr CreateBasicMessage() {
+    application_manager::MessagePtr message = utils::MakeShared<Message>(
+        MessagePriority::FromServiceType(protocol_handler::ServiceType::kRpc));
+    message->set_connection_key(kConnectionKey);
+    message->set_function_id(RCFunctionID::SET_INTERIOR_VEHICLE_DATA);
+    message->set_function_name("SetInteriorVehicleData");
+    return message;
+  }
+
+ protected:
+  utils::SharedPtr<NiceMock<application_manager::MockService> > mock_service_;
+  utils::SharedPtr<NiceMock<MockApplication> > mock_app_;
+  utils::SharedPtr<remote_control::RCAppExtension> rc_app_extention_;
+  testing::NiceMock<remote_control_test::MockRemotePluginInterface>
+      mock_module_;
+  testing::NiceMock<remote_control_test::MockResourceAllocationManager>
+      mock_allocation_manager_;
+  RemotePluginInterface::RCPluginEventDispatcher event_dispatcher_;
+  const uint32_t app_id_ = 11u;
+};
+
+TEST_F(SetInteriorVehicleDataRequestTest,
+       Execute_ValidMessageWithoutReadOnlyParams) {
+  // Arrange
+  application_manager::MessagePtr mobile_message = CreateBasicMessage();
+  mobile_message->set_json_message(kValidMobileRequestWithOnlySettableParams);
+  // Expectations
+  EXPECT_CALL(mock_module_, service())
+      .Times(2)
+      .WillRepeatedly(Return(mock_service_));
+  EXPECT_CALL(*mock_service_, GetApplication(mobile_message->connection_key()))
+      .WillOnce(Return(mock_app_));
+  EXPECT_CALL(*mock_service_, ValidateMessageBySchema(*mobile_message))
+      .WillOnce(Return(application_manager::MessageValidationResult::SUCCESS));
+  EXPECT_CALL(*mock_service_, CheckPolicyPermissions(mobile_message))
+      .WillOnce(Return(mobile_apis::Result::eType::SUCCESS));
+  application_manager::AppExtensionPtr invalid_ext;
+  EXPECT_CALL(*mock_app_, QueryInterface(kModuleId))
+      .WillOnce(Return(invalid_ext))
+      .WillRepeatedly(Return(rc_app_extention_));
+  application_manager::AppExtensionPtr app_extension;
+  EXPECT_CALL(*mock_app_, AddExtension(_))
```

Comment:
@Itileda this part of unit test is the crux part of UT. please emphasize it with spaces